src/modules/interface.js

```diff
@@ -0,0 +1,236 @@
+      optionBtn.forEach((btn) => {
+        btn.addEventListener('click', (event) => {
+          const listItem = event.target.closest('.toDoList');
+          if (Interface.editList === '') {
+            Interface.editList = listItem;
+            Interface.editTask(Interface.editList);
+          } else if (Interface.editList) {
+            if (Interface.editList !== listItem) {
+              Interface.cancelEditTask(Interface.editList);
+              Interface.editList = listItem;
+              Interface.editTask(Interface.editList);
+            } else if (Interface.editList === listItem) {
+              Interface.cancelEditTask(listItem);
+              Interface.editTask(Interface.editList);
+            }
+          }
+        });
+      });
+
+      // add event listener to each checkbox
+      const checkboxes = document.querySelectorAll('.checkbox');
+      checkboxes.forEach((checkbox) => {
+        checkbox.addEventListener('change', (event) => {
+          Interface.lineOnTask(event.target);
+          const task = Interface.toDoList.find(
+            (task) => task.index === parseInt(event.target.id, 10),
+          );
+          task.done = event.target.checked;
+          Storage.updateTasks(Interface.toDoList);
+        });
+      });
+    }
+  };
+
+  // add task to list
+  static addTask = (e) => {
+    if (e.key === 'Enter') {
+      e.preventDefault();
+      const title = document.querySelector('#title').value;
+      if (title === '') {
+        Interface.showAlert('Please fill emepty fields!', 'danger');
+      } else {
+        const index = Interface.toDoList.length;
+        const task = new ToDo(title, false, index + 1);
+        Storage.setTasks(task);
+        Interface.toDoList = Storage.getTasks();
+        Interface.completedTasks = [];
+        Interface.displayToDoList();
+        Interface.emptyFields();
+      }
+    }
+  };
+
+  /// clear fields
+  static emptyFields = () => {
+    document.querySelector('#title').value = '';
+  };
+
+  // reorder list
+  static arrangeList = () => {
+    const newTasks = Interface.toDoList;
+    let currentIndex = 1;
+    newTasks.forEach((task) => {
+      task.index = currentIndex;
+      currentIndex += 1;
+    });
+    Storage.updateTasks(newTasks);
+  };
+
+    // clear all completed tasks
+    static clearAllTasks = (e) => {
+      e.preventDefault();
+      if (Interface.completedTasks.length === 0) {
+        Interface.showAlert('No task selected!', 'danger');
+      } else {
+        const i = 0;
+        while (i < Interface.completedTasks.length) {
+          const completedTask = Interface.completedTasks[i];
+          Storage.clearTasks(completedTask);
+          Interface.toDoList = Storage.getTasks();
+          Interface.completedTasks.splice(i, 1); // Remove the completed task from the array
+        }
+        Interface.arrangeList();
+        Interface.displayToDoList();
+      }
+    };
+
+  // strike through list if checked
```

Comment:
- [ ] As suggested by the previous code reviewer, a user is unable to undo the completing of a task after it it marked as complete. please kindly fix that.